### PR TITLE
lint: use precomputed imports map in importShadow

### LINKED ISSUE #532: lint: handle renamed imports more efficiently
Looking at `importShadow`, it's clear that the current way of handling renamed imports is rather inefficient and somewhat clumsy. We need something better that does not require defs traversal per definition (close to n^2).

### PR DESCRIPTION
**Pull request by Quasilyte**
Fixes #532

### CODE REVIEW

File: lint/lint_private.go [hunk 1]
```diff
@@ -193,3 +194,17 @@ func addChecker(c abstractChecker, attrs ...checkerAttribute) {
 	}
 	checkerPrototypes[rule.name] = proto
 }
+
+func resolvePkgObjects(ctx *context, f *ast.File) {
+	ctx.pkgObjects = make(map[*types.PkgName]string)
```
Comment (cristaloleg) [inline lint/lint_private.go]: `make(map[*types.PkgName]string, len(f.Imports))` ?
Just to make it explicit what we're expecting, unless you're against this.

File: lint/lint.go [hunk 2]
```diff
@@ -123,6 +126,19 @@ type Context struct {
...
 	// Arch-dependent.
 	sizesInfo types.Sizes
+
+	// require records what optional resources are required
+	// by the checkers set that use this context.
+	//
+	// Every require fields makes associated context field
+	// to be properly initialized.
+	// For example, Context.require.pkgObjects => Context.pkgObjects.
+	require struct {
```
Comment (cristaloleg) [inline lint/lint.go]: Damn, that's a good pattern!

File: lint/lint_private.go (hunk 1 above)
Comment (Quasilyte) [reply lint/lint_private.go]: OK.

**Comment by Quasilyte** [comment]
We definitely don't want to  disable `unused` from CI pipeline, 
so I think we can afford some workarounds in this case.

**Pull request merged**